lib/NativeCall.rakumod

```diff
@@ -260,6 +263,144 @@ sub resolve-libname($libname) {
     $libname.platform-library-name.Str
 }
 
+use MONKEY-SEE-NO-EVAL;
+my $use-dispatcher = so $*RAKU.compiler.?supports-op('dispatch_v') && EVAL q:to/CODE/;
+    use nqp;
+    sub raku-native-dispatch-deproxy($callee, *@args) {
+        $callee(|@args) # implicitly fetches values from proxies
+    }
+    my &raku-nativecall := -> $capture is raw {
+        my $track_callee := nqp::dispatch('boot-syscall', 'dispatcher-track-arg', $capture, 0);
+        nqp::dispatch('boot-syscall', 'dispatcher-guard-literal', $track_callee);
+        my $callee := nqp::captureposarg($capture, 0);
+        $callee.setup;
+
+        my Mu $args := nqp::dispatch('boot-syscall', 'dispatcher-drop-arg', $capture, 0);
+        my $signature := nqp::getattr($callee, Code, '$!signature');
+        my int $readonly = nqp::getattr_i($signature, Signature, '$!readonly');
+        my int $pos-args = nqp::captureposelems($args);
+        my int $i = 0;
+        my int $continue = 1;
```

Comment:
I wonder if `$no-proxy` would be more descriptive, given that's what this is for.